src/components/switchTheme/useSwitchTheme.tsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react';
+
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+interface IProps {
+  children: ReactNode;
+}
+
+export const ThemeContext = createContext<ThemeContextType | null>(null);
+
+const ThemeContextProvider = ({ children }: IProps) => {
+  const getInitialTheme = () => {
+    const savedTheme = localStorage.getItem('theme');
+    if (savedTheme) {
+      return savedTheme as 'light' | 'dark';
+    }
+
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    return prefersDark ? 'dark' : 'light';
+  };
+
+  const [theme, setTheme] = useState<'light' | 'dark'>(getInitialTheme);
+
+  const toggleTheme = useCallback(() => {
+    setTheme((prev) => {
+      const newTheme = prev === 'light' ? 'dark' : 'light';
+      document.documentElement.setAttribute('data-theme', newTheme);
+      localStorage.setItem('theme', newTheme);
+      return newTheme;
+    });
+  }, []);
```

Comment:
Why do you use useCallback here?

Reply:
frankly speaking we dont need useCallback here, as toggleTheme doesn't depend on any external variables that could change.